src/plugins/intel_gpu/src/kernel_selector/core/cl_kernels/matrix_nms_ref.cl

```diff
@@ -0,0 +1,329 @@
+// Copyright (C) 2018-2022 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "include/batch_headers/common.cl"
+#include "include/batch_headers/data_types.cl"
+
+#define NUM_BATCHES INPUT0_BATCH_NUM
+#define NUM_BOXES   INPUT0_FEATURE_NUM
+#define NUM_CLASSES INPUT1_FEATURE_NUM
+
+typedef struct {
+    int batch_idx;
+    int class_idx;
+    int box_idx;
+    INPUT1_TYPE score;
+} FUNC(BoxInfo);
+
+#define BOX_INFO FUNC(BoxInfo)
+
+#define unroll_for __attribute__((opencl_unroll_hint)) for
+
+inline INPUT1_TYPE FUNC(decay_gaussian)(INPUT1_TYPE iou, INPUT1_TYPE max_iou) {
+    return exp((max_iou * max_iou - iou * iou) * GAUSSIAN_SIGMA);
+}
+
+inline INPUT1_TYPE FUNC(decay_linear)(INPUT1_TYPE iou, INPUT1_TYPE max_iou) {
+    return (INPUT1_VAL_ONE - iou) / (INPUT1_VAL_ONE - max_iou + TINY);
```

Comment:
is this "TINY" to avoid division by zero?

Reply:
for me EPS is more clear, but if term TINY is already used- it's fine

Reply:
Agreed with tgubanova-lohika, EPS is more familiar term in openvino.